src/kernel.cr

```diff
@@ -25,15 +21,17 @@
   #
   # Typically used to output data and information.
   #
-  # NOTE: Gets flushed when a newline is written.
-  STDOUT = IO::FileDescriptor.from_stdio(1).tap { |f| f.flush_on_newline = true }
+  # If it is a TTY device, `sync` will be true for this it
```

Comment:
```suggestion
  # When this is a TTY device, no write buffering will be done.
```
The line below this one can be removed. Ditto below.

Reply:
I changed the text a bit but:
- changing `sync` to `no write buffering` is wrong because what's really happening is that `sync` becomes `true` and you can always change that (if we say "no write buffering" it's not clear how to revert it)
- this check happens at the start of the program so you can revert it, and I think it's important to say this

Reply:
I agree but shouldn't `sync` be maybe `sync?`? `sync` does not exist.

Reply:
there's `sync=` and `sync?`. In any case this doesn't matter much, eventually we should rename `sync` to `write_buffering`.